.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,20 @@
+name: Publish Package to npmjs
+
+on:
+  push:
+    branches: 
+    - main
+
+jobs:
+  build-publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
A couple of weeks ago `actions/checkout@v4` came out. I think we should be able to use that one without issue :). Otherwise it looks great!

Reply:
Good point. updated.